creativecloud/scripts/mep/aexg4866/interactive-marquee/interactive-marquee.js

```diff
@@ -0,0 +1,113 @@
+      sib.classList.add(className);
+      sib.previousElementSibling?.classList.add('icon-area');
+    }
+    const iconAreaElements = el.querySelector('.icon-area');
+    const iconText = createTag('div', { class: `heading-${typeSize[3]} icon-text` });
+    iconAreaElements.appendChild(iconText);
+    iconAreaElements?.classList.add('icon-area');
+    iconText.innerText = (iconAreaElements.textContent.trim());
+    iconText.previousSibling.textContent = '';
+    if (isMwebMarquee) {
+      const foreground = el.closest('.foreground');
+      const mwebContainer = createTag('div', { class: 'mweb-container' });
+      const actionItem = el.querySelector('.action-area');
+      mwebContainer.append(heading.cloneNode(true), actionItem.cloneNode(true));
+      heading.classList.add('mobile-cta-top');
+      actionItem.classList.add('mobile-cta-top');
+      if (sib) {
+        mwebContainer.prepend(sib.cloneNode(true));
+        sib.classList.add('mobile-cta-top');
+      }
+      foreground.prepend(mwebContainer);
+    }
+  };
+  decorate(heading, config);
+}
+
+function extendButtonsClass(text) {
+  const buttons = text.querySelectorAll('.con-button');
+  if (buttons.length === 0) return;
+  buttons.forEach((button) => { button.classList.add('button-justified-mobile'); });
+}
+
+function interactiveInit(el, decorateButtons, decorateBlockBg, createTag) {
+  const isLight = el.classList.contains('light');
+  if (!isLight) el.classList.add('dark');
+  const children = el.querySelectorAll(':scope > div');
+  const foreground = children[children.length - 1];
+  if (children.length > 1) {
+    children[0].classList.add('background');
+    decorateBlockBg(el, children[0], { useHandleFocalpoint: true });
+  }
+  foreground.classList.add('foreground', 'container');
+  const headline = foreground.querySelector('h1, h2, h3, h4, h5, h6');
+  const text = headline.closest('div');
+  text.classList.add('text');
+  const mediaElements = foreground.querySelectorAll(':scope > div:not([class])');
+  const media = mediaElements[0];
+  if (media) {
+    const interactiveBox = createTag('div', { class: 'interactive-container' });
+    mediaElements.forEach((mediaDiv) => {
+      mediaDiv.classList.add('asset');
+      interactiveBox.appendChild(mediaDiv);
+    });
+    foreground.appendChild(interactiveBox);
+  }
+
+  const firstDivInForeground = foreground.querySelector(':scope > div');
+  if (firstDivInForeground?.classList.contains('asset')) el.classList.add('row-reversed');
+  const isMwebMarquee = el.classList.contains('mobile-cta-top');
+
+  decorateButtons(text, 'button-xl');
+  decorateText(text, createTag, isMwebMarquee);
+  extendButtonsClass(text);
+}
+
+export default async function init(el) {
+  const miloLibs = getLibs('/libs');
+  const { decorateButtons, decorateBlockBg } = await import(`${miloLibs}/utils/decorate.js`);
+  const { createTag, loadStyle } = await import(`${miloLibs}/utils/utils.js`);
+  switch (true) {
```

Comment:
Since you've removed most of the cases, converting the switch statement to an if else might be clearer, but not change necessary.

Reply:
Noted, and discussed during review.